### PR TITLE
Rework check-annotations in prevision to more checks + add it to CI

### LINKED ISSUE #39: Run `./bin/check-annotations` as CI (or similar)
#38 introduced `./bin/check-annotations` which helps identify problems with the contents of `annotations.tsv`. This highlights issues that may only become evident stochastically after running the auspice build. 

It would be nice to run this as a CI (or similar). For this to happen the script would have to be modified to exit appropriately which it doesn't currently do.

### PR DESCRIPTION
**Pull request by jstoja**
### Description of proposed changes    
Rework `bin/check-annotations` to make it more modular and be able to add more checks later on.
I've also added it to CI as mentioned in the original issue.

### Related issue(s)  
Fixes #39 

### Testing
The CI should run the script. I've checked that it reports error with the annotation file before #38 .

### CODE REVIEW

**Comment by jstoja** [comment]
The pipeline fails with the following error:
`./bin/fetch-from-gisaid: line 4: GISAID_API_ENDPOINT: The GISAID_API_ENDPOINT environment variable is required.`

I'm not sure this is because of my changes since `./bin/fetch-from-gisaid` is called after the script changed in this PR (`./bin/check-annotations`).

**Comment by jameshadfield** [comment]
Thanks @jstoja! I'll let @kairstenfay and @tsibley review the CI approach.

File: .github/workflows/ingest.yml [hunk 1]
```diff
@@ -21,6 +21,7 @@ jobs:
         python3 -m pip install --upgrade pip setuptools
         python3 -m pip install pipenv
         pipenv sync
+        pipenv run ./bin/check-annotations
```
[the remark this answers is not in the document]
Comment (kairstenfay) [reply .github/workflows/ingest.yml:24]: Just checking -- @tsibley does it make sense to only validate annotations if we're on the master branch?

File: bin/check-annotations [hunk 2]
```diff
@@ -4,45 +4,108 @@ Check annotations for potential errors / missing data
 """
 
 ## Not part of this script, but you can check if a strain has duplica
-# 
 
-import pandas as pd
 import sys
+import pandas as pd
+
+ANNOTATIONS_FILE = "./source-data/annotations.tsv"
```
Comment (kairstenfay) [inline bin/check-annotations:11]: Before merging this, we'll want to stop relying on a file path relative to this script but instead relative to the base of this repo. e.g like what we've done in [transform](https://github.com/nextstrain/ncov-ingest/blob/6d4b6de889fbc54a3cba220cf4b813587242fc60/bin/transform#L251).

File: .github/workflows/ingest.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (jstoja) [reply .github/workflows/ingest.yml:24]: The only downside that I see if this is checked only on master is if contributors modify the annotations file in PRs then it wouldn't be checked until the PR is merged in master.
In addition, correct me if I'm wrong, but the `annotations.tsv` file is used by the transform script, which means that we would want to check if *before* the transform operation, right? Shouldn't we validate that the file is correct even before starting any operation in `ingest`?

File: bin/check-annotations (hunk 2 above)
Comment (jstoja) [reply bin/check-annotations]: Done! I've also implemented this with argparse (like in transform), so it's easier to specify a different file name (helped with testing).

File: .github/workflows/ingest.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (kairstenfay) [reply .github/workflows/ingest.yml]: > The only downside that I see if this is checked only on master is if contributors modify the annotations file in PRs then it wouldn't be checked until the PR is merged in master.

Good point @jstoja. We would want to check for errors in `annotations.tsv` by running it as part of the CI on open pull requests.


> In addition, correct me if I'm wrong, but the `annotations.tsv` file is used by the transform script, which means that we would want to check if _before_ the transform operation, right? Shouldn't we validate that the file is correct even before starting any operation in `ingest`?

Another good catch :slightly_smiling_face:. Yes I overlooked the fact that `annotations.tsv` is used in the `transform` script to modify the metadata. Indeed if the annotations were bad, we wouldn't want to even bother with fetching data from GISAID at all (first step of the `ingest` process). @tsibley was the one to initially refactor the github workflow (CI) into its own script (`ingest`), so I welcome him to opine here.

File: bin/check-annotations (hunk 2 above)
Comment (kairstenfay) [reply bin/check-annotations]: Awesome. Thanks for doing this.

File: .github/workflows/ingest.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tsibley) [reply .github/workflows/ingest.yml]: :+1: for running it on branches/PRs too and running it before `transform`. Even before `fetch-from-gisaid` would also be fine. I agree it should be run from `ingest` so its part of the process when run manually (not via CI).

[the remark this answers is not in the document]
Comment (kairstenfay) [reply .github/workflows/ingest.yml]: Thanks, Tom. I'll push a new commit running `check-annotations` as the first step in `ingest` then merge.

**Pull request merged**